tests/components/minio/test_minio.py

```diff
@@ -1,44 +1,195 @@
 """Tests for Minio Hass related code."""
-import homeassistant.core as ha
-import homeassistant.components.minio as minio
+import asyncio
+import json
+from unittest.mock import MagicMock, patch
 
+from homeassistant.components.minio import QueueListener, DOMAIN, CONF_HOST, \
+    CONF_PORT, CONF_ACCESS_KEY, CONF_SECRET_KEY, CONF_SECURE, CONF_LISTEN, \
+    CONF_LISTEN_BUCKET
+from homeassistant.const import EVENT_HOMEASSISTANT_START, \
+    EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import callback
+from homeassistant.setup import async_setup_component
 
-async def test_queue_listener(hass):
-    """Tests QueueListener firing events on Hass event bus."""
-    queue_listener = minio.QueueListener(hass)
-    queue_listener.start()
+from tests.components.minio.common import TEST_EVENT
+
+
+async def test_minio_services(hass, caplog):
+    """Test Minio services."""
+    hass.config.whitelist_external_dirs = set('/tmp')
+
+    minio_client_mock = MagicMock()
+
+    with patch('homeassistant.components.minio.Minio') as minio_mock:
+        minio_mock.return_value = minio_client_mock
+
+        await async_setup_component(
+            hass,
+            DOMAIN,
+            {
+                DOMAIN: {
+                    CONF_HOST: 'localhost',
+                    CONF_PORT: '9000',
+                    CONF_ACCESS_KEY: 'abcdef',
+                    CONF_SECRET_KEY: '0123456789',
+                    CONF_SECURE: 'true',
+                }
+            }
+        )
+
+        hass.bus.fire(EVENT_HOMEASSISTANT_START)
```

Comment:
Use the async api in async context. We're in a coroutine here, ie async context.

We can await `hass.async_start` instead.